internal/weapons/catalyst/moonglow/moonglow.go

```diff
@@ -0,0 +1,65 @@
+package moonglow
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterWeaponFunc("everlastingmoonglow", weapon)
+	core.RegisterWeaponFunc("everlasting moonglow", weapon)
+	core.RegisterWeaponFunc("moonglow", weapon)
+}
+
+func weapon(char core.Character, c *core.Core, r int, param map[string]int) string {
+	mheal := make([]float64, core.EndStatType)
+	mheal[core.Heal] = 0.075 + float64(r)*0.025
+	char.AddMod(core.CharStatMod{
+		Key: "moonglow-heal-bonus",
+		Amount: func() ([]float64, bool) {
+			return mheal, true
+		},
+		Expiry: -1,
+	})
+
+	matk := make([]float64, core.EndStatType)
+	matk[core.ATK] = (0.0005 + float64(r)*0.0005) * char.MaxHP()
```

Comment:
i don't think anything exists in game right now that modify maxhp in combat but to be future proof probably best moving this line into the Amount function?

i would also move `atk.Info.AttackTag == core.AttackTagNormal` into an if and return nil if not true so that way the maxhp calculation doesn't get called if not required (after moving it into the amount function that is)